weave

```diff
@@ -69,6 +76,10 @@ command_exists () {
     command -v $1 >/dev/null 2>&1
 }
 
+log()   { echo "$1"; }
+warn()  { log "WARNING: $1"; }
+fatal() { log "ERROR: $1"; exit 1; }
+
```

Comment:
The majority of log messages should (and, in the original, used to) go to stderr, not stdout.